File: Java-Python-SQL-Fundamentals/Java Fundamentals/Java_Inheritance_Prac5/src/main/java/org/example/Account.java
package org.example;


public class Account {

    private String username, email;
    private boolean isLoggedIn;


    public Account(String UserName, String mail) {

        this.username = UserName;
        this.email = mail;
        isLoggedIn = false;
    }

    public void login() {

        if (isLoggedIn) {
            System.out.println("Already logged in.");
            return;

        }

        System.out.println("Logging in... \nWelcome " + username);
        isLoggedIn = true;

    }

    public void logout() {
        if (!isLoggedIn) {
            System.out.println("Already logged out!");
            return;

        }

        System.out.println("Logged out successfully");
        isLoggedIn = false;

    }

    protected boolean isUserLoggedIn() {
        return isLoggedIn;

    }

    public String getAccountInfo() {

        return "Account details: " +
                "\nUsername: " + username +
                "\nEmail: " + email +
                "\nLogin Status: " + isLoggedIn +
                "\n";

    }


}

File: Java-Python-SQL-Fundamentals/Python-Fundamentals/Python_inheritance_practice1/Account.py
class Account:

    def __init__(self, username, email_address):
        self.name = username
        self.email = email_address
        self.logged_in = False

    def login(self):

        if self.logged_in:
            print("Already Logged in")
            return

        print("Logging in....")
        print(f"Welcome {self.name}\n")
        self.logged_in = True

    def logout(self):

        if not self.logged_in:
            print("Not logged in")
            return

        print("Logging out")
        self.logged_in = False

    def _is_user_logged_in(self):
        return self.logged_in

    def get_account_info(self):
        return ('Account Details:\n'
                f'Username: {self.name}\n'
                f'Email: {self.email}\n'
                f'Login Status: {self.logged_in}')
